osu.Game.Rulesets.Taiko/Difficulty/TaikoDifficultyCalculator.cs

```diff
@@ -48,47 +47,29 @@ public TaikoDifficultyCalculator(IRulesetInfo ruleset, IWorkingBeatmap beatmap)
 
         protected override IEnumerable<DifficultyHitObject> CreateDifficultyHitObjects(IBeatmap beatmap, double clockRate)
         {
-            List<DifficultyHitObject> taikoDifficultyHitObjects = new List<DifficultyHitObject>();
-
-            for (int i = 2; i < beatmap.HitObjects.Count; i++)
-            {
-                taikoDifficultyHitObjects.Add(
-                    new TaikoDifficultyHitObject(
-                        beatmap.HitObjects[i], beatmap.HitObjects[i - 1], beatmap.HitObjects[i - 2], clockRate, taikoDifficultyHitObjects, taikoDifficultyHitObjects.Count
-                    )
-                );
-            }
-
-            return taikoDifficultyHitObjects;
+            return TaikoDifficultyPreprocessor.CreateDifficultyHitObjects(beatmap, clockRate);
         }
 
         protected override DifficultyAttributes CreateDifficultyAttributes(IBeatmap beatmap, Mod[] mods, Skill[] skills, double clockRate)
         {
             if (beatmap.HitObjects.Count == 0)
                 return new TaikoDifficultyAttributes { Mods = mods };
 
-            var colour = (Colour)skills[0];
-            var rhythm = (Rhythm)skills[1];
-            var stamina = (Stamina)skills[2];
+            var combined = (Peaks)skills[0];
 
-            double colourRating = colour.DifficultyValue() * colour_skill_multiplier;
-            double rhythmRating = rhythm.DifficultyValue() * rhythm_skill_multiplier;
-            double staminaRating = stamina.DifficultyValue() * stamina_skill_multiplier;
+            double colourRating = combined.ColourDifficultyValue * difficulty_multiplier;
+            double rhythmRating = combined.RhythmDifficultyValue * difficulty_multiplier;
+            double staminaRating = combined.StaminaDifficultyValue * difficulty_multiplier;
 
-            double staminaPenalty = simpleColourPenalty(staminaRating, colourRating);
-            staminaRating *= staminaPenalty;
+            double combinedRating = combined.DifficultyValue() * difficulty_multiplier;
+            double starRating = rescale(combinedRating * 1.4);
 
-            //TODO : This is a temporary fix for the stamina rating of converts, due to their low colour variance.
-            if (beatmap.BeatmapInfo.Ruleset.OnlineID == 0 && colourRating < 0.05)
+            // TODO: This is temporary measure as we don't detect abuse-type playstyles of converts within the current system.
```

Comment:
What abuse-type playstyles? Would strongly recommend leaving more detail here to give the opportunity for anyone to understand the issue in case they want to attempt a solution.

Reply:
This nerf has been in here for a while now in numerous different forms, due to our system not detecting actual cheating methods (we label them as abuse-type playstyles which is generally in line with the consensus of macros and touchscreens within the taiko community). However if these scores can be removed this is more than able to be removed.

Reply:
This is referring to unreasonably high speed mono, which is completely impossible to play for 2 fingers (anywhere from 700bpm to 1500+bpm 1/4 mono streams). But the assumption of 2 fingers of are false at pure mono patterns, due to the player being able to press the same key using multiple finger (move the other hand over), or even add inputs using mouse clicks. In the future there are plans to mark objects as multi-input-able (or whatever name makes more sense) and increasing the finger count to a higher number for those.